arch/xtensa/src/esp32s3/Kconfig

```diff
@@ -1090,9 +1090,12 @@ choice ESP32S3_SPIRAM_HEAP
 
 config ESP32S3_SPIRAM_COMMON_HEAP
 	bool "Additional region to kernel heap"
+	depends on !ESP32S3_SPIFLASH && !ESPRESSIF_WIFI && !BUILD_PROTECTED
```

Comment:
isn't CONFIG_XTENSA_IMEM_USE_SEPARATE_HEAP enough for wifi?

i guess it also makes sense to allocate LPWORK stack from the internal memory. what do you think?

Reply:
>  I'd stick with kernel/user heap separation in this case and remove CONFIG_XTENSA_IMEM_USE_SEPARATE_HEAP for ESP32-S3 in the future.

i didn't know you wanted to deprecate CONFIG_XTENSA_IMEM_USE_SEPARATE_HEAP.

i always think nuttx should have something like esp-idf's heap caps.
for small devices, separate heaps are often too wasteful wrt space.

> Please note that for ESP32-S3 there are no limitations on accessing PSRAM's through the instruction bus (something that is true for ESP32).

i know. my configuration uses it for text heap. (text segment of wamr aot modules)

> Please check https://github.com/apache/nuttx/pull/16756#issuecomment-3113617106

i will reply there.